File: leetcode-04/solutions/python3/0414.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        list = []

        for num in set(nums):
            bisect.insort(list, -num)
            del list[3:]

        return -list[2] if len(list) > 2 else -list[0]

File: leetcode-04/solutions/cpp/0414.cpp
class Solution {
 public:
  int thirdMax(vector<int>& nums) {
    set<int> set;

    for (int num : nums) {
      set.insert(num);
      if (set.size() > 3) set.erase(set.begin());
    }

    return set.size() == 3 ? *set.begin() : *set.rbegin();
  }
};